libminifi/include/Repository.h

```diff
@@ -0,0 +1,294 @@
+				{
+				}
+			}
+			logger_->log_info("NiFi Provenance Max Storage Time: [%d] ms", _maxPartitionMillis);
+			leveldb::Options options;
+			options.create_if_missing = true;
+			leveldb::Status status = leveldb::DB::Open(options, _directory.c_str(), &_db);
+			if (status.ok())
+			{
+				logger_->log_info("NiFi Provenance Repository database open %s success", _directory.c_str());
+			}
+			else
+			{
+				logger_->log_error("NiFi Provenance Repository database open %s fail", _directory.c_str());
+				return false;
+			}
+		}
+
+		if (_type == FLOWFILE)
+		{
+			if (!(configure_->get(Configure::nifi_flowfile_repository_enable, value)
+					&& StringUtils::StringToBool(value, _enable))) {
+				_enable = true;
+			}
+			if (!_enable)
+				return false;
+			if (configure_->get(Configure::nifi_flowfile_repository_directory_default, value))
+			{
+				_directory = value;
+			}
+			logger_->log_info("NiFi FlowFile Repository Directory %s", _directory.c_str());
+			if (configure_->get(Configure::nifi_flowfile_repository_max_storage_size, value))
+			{
+				Property::StringToInt(value, _maxPartitionBytes);
+			}
+			logger_->log_info("NiFi FlowFile Max Partition Bytes %d", _maxPartitionBytes);
+			if (configure_->get(Configure::nifi_flowfile_repository_max_storage_time, value))
+			{
+				TimeUnit unit;
+				if (Property::StringToTime(value, _maxPartitionMillis, unit) &&
+							Property::ConvertTimeUnitToMS(_maxPartitionMillis, unit, _maxPartitionMillis))
+				{
+				}
+			}
+			logger_->log_info("NiFi FlowFile Max Storage Time: [%d] ms", _maxPartitionMillis);
+			leveldb::Options options;
+			options.create_if_missing = true;
+			leveldb::Status status = leveldb::DB::Open(options, _directory.c_str(), &_db);
+			if (status.ok())
+			{
+				logger_->log_info("NiFi FlowFile Repository database open %s success", _directory.c_str());
+			}
+			else
+			{
+				logger_->log_error("NiFi FlowFile Repository database open %s fail", _directory.c_str());
+				return false;
+			}
+		}
+
+		return true;
+	}
+	//! Put
+	virtual bool Put(std::string key, uint8_t *buf, int bufLen)
+	{
+		if (!_enable)
+			return false;
+			
+		// persistent to the DB
+		leveldb::Slice value((const char *) buf, bufLen);
+		leveldb::Status status;
+		status = _db->Put(leveldb::WriteOptions(), key, value);
+		if (status.ok())
+			return true;
+		else
+			return false;
+	}
+	//! Delete
+	virtual bool Delete(std::string key)
+	{
+		if (!_enable)
+			return false;
+		leveldb::Status status;
+		status = _db->Delete(leveldb::WriteOptions(), key);
+		if (status.ok())
+			return true;
+		else
+			return false;
+	}
+	//! Get
+	virtual bool Get(std::string key, std::string &value)
+	{
+		if (!_enable)
+			return false;
+		leveldb::Status status;
+		status = _db->Get(leveldb::ReadOptions(), key, &value);
+		if (status.ok())
+			return true;
+		else
+			return false;
+	}
+	//! destroy
+	void destroy()
+	{
+		if (_db)
```

Comment:
we only call destroy when we delete the repo, will change to a private function.